File: backend/keyManager.js
require("dotenv").config({
  quiet: true
});

const keys = Object.keys(process.env)
  .filter(key => key.startsWith("GOOGLE_API_KEY_"))
  .sort()
  .map(key => process.env[key]);

  if (keys.length === 0) {
  throw new Error(
    "No Google API keys found in .env file"
  );
}

let currentIndex = 0;

function getKey() {
  return keys[currentIndex];
}

function rotateKey() {
  currentIndex = (currentIndex + 1) % keys.length;
}

function totalKeys() {
  return keys.length;
}

module.exports = {
  getKey,
  rotateKey,
  totalKeys
};

File: backend/keyManager.py
from dotenv import load_dotenv
import os

# Load environment variables
load_dotenv()

# Read all Google API keys from .env
keys = sorted(
    value
    for key, value in os.environ.items()
    if key.startswith("GOOGLE_API_KEY_")
)

if not keys:
    raise ValueError("No Google API keys found in .env file")

current_index = 0


def getKey():
    """Return the current API key."""
    return keys[current_index]


def rotateKey():
    """Rotate to the next API key."""
    global current_index
    current_index = (current_index + 1) % len(keys)


def totalKeys():
    """Return the total number of API keys."""
    return len(keys)